### PR TITLE
V3 visualizer support

### PR DESCRIPTION
**Pull request by moratom**
This PR adds the following:
* Protobuf serialization for a subset of DAI messages
* RemoteConnection which exposes DAI streams over a WebSocket
* Serving of the embedded client for the RemoteConnection
* Initial integration of `vcpkg` into DAI

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -13,11 +30,42 @@ option(DEPTHAI_ENABLE_LIBUSB "Enable usage of libusb and interaction with USB de
...
+# Check if on 32 bit linux - default without CURL support
+if(CMAKE_SIZEOF_VOID_P EQUAL 4 AND UNIX)
+    set(DEPTHAI_DEFAULT_CURL_SUPPORT OFF)
+else()
+    set(DEPTHAI_DEFAULT_CURL_SUPPORT ON)
+endif()
+
+option(DEPTHAI_ENABLE_CURL "Enable CURL support" ${DEPTHAI_DEFAULT_CURL_SUPPORT})
+
+if(DEPTHAI_ENABLE_CURL)
```
[the remark this answers is not in the document]
Comment (moratom) [reply CMakeLists.txt:48]: Yes, it only has to be added before the `project()` call.

You mean a macro for adding the vcpkg packages?

File: include/depthai/pipeline/datatype/EncodedFrame.hpp [hunk 2]
```diff
@@ -171,6 +172,22 @@ class EncodedFrame : public Buffer {
      */
     EncodedFrame& setProfile(Profile profile);
 
+#ifdef DEPTHAI_ENABLE_PROTOBUF
```
[the remark this answers is not in the document]
Comment (moratom) [reply include/depthai/pipeline/datatype/EncodedFrame.hpp:175]: Every serializable class needs to override the functions with their own implementation, that's why the declarations need to be in every header that implements the proto interface.

**Review by asahtik — COMMENTED** [review]
Does this build on RVC2/4?

File: src/utility/ProtoSerialize.cpp [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    if(descriptor == nullptr) {
+        throw std::runtime_error("Failed to get protobuf descriptor");
+    }
+    ProtoSerializable::SchemaPair returnPair;
+    returnPair.schemaName = descriptor->full_name();
+    returnPair.schema = serializeFdSet(descriptor);
+    return returnPair;
+}
+
+void serializeImgTransormation(proto::common::ImgTransformation* imgTransformation, const ImgTransformation& transformation) {
```
Comment (asahtik) [inline src/utility/ProtoSerialize.cpp:56]: serializeImgTrans**f**ormation

**Review by asahtik — APPROVED** [review]

**Pull request merged**